dll/cpl/sysdm/smbios.c

```diff
@@ -76,6 +77,35 @@ TrimPunctuation(
     }
 }
 
+/*
+ * Case insensitive variant of strstr
+ */
+static
+CHAR * stristr(const CHAR *s, const CHAR *b)
```

Comment:
In the places where you use such function, you keep performing case insensitive comparisons or searches.
This is definitely under optimized, and thus, you are forced to implement such a function.
I'd rather see you upcase one for all the string you search and the SMBIOS output string so that your code will perform better.